qiskit/circuit/instruction.py

```diff
@@ -275,13 +281,36 @@ def assemble(self):
             instruction.qubits = list(range(self.num_qubits))
         if self.num_clbits:
             instruction.memory = list(range(self.num_clbits))
+        # Add label if defined
+        if self.label:
+            instruction.label = self.label
         # Add condition parameters for assembler. This is needed to convert
         # to a qobj conditional instruction at assemble time and after
         # conversion will be deleted by the assembler.
         if self.condition:
             instruction._condition = self.condition
         return instruction
 
+    @property
+    def label(self) -> str:
+        """Return instruction label"""
+        return self._label
+
+    @label.setter
+    def label(self, name: str):
+        """Set instruction label to name
+
+        Args:
+            name (str or None): label to assign instruction
+
+        Raises:
+            TypeError: name is not string or None.
+        """
+        if isinstance(name, (str, type(None))):
+            self._label = name
```

Comment:
Not necessarily something to change, but I'm a bit confused why pylint doesn't complain about `self._label` not being defined in init. Maybe that's due to the workaround until Aer's next release, which may define `self._label`.

Reply:
My best guess is that because `self._label` might be set (on L87), pylint doesn't consider this an issue.